storage/tianmu/core/rc_attr.cpp

```diff
@@ -1,1429 +1,1436 @@
-  if (std::find(keycols.begin(), keycols.end(), col) == keycols.end())
-    return;
-
-  if (!v.HasValue())
-    throw common::Exception("primary key not support null!");
-
-  if (GetPackType() == common::PackType::STR) {
-    auto &vnew = v.GetString();
-    auto vold = GetValueString(row);
-    std::string_view nkey(vnew.data(), vnew.length());
-    std::string_view okey(vold.val_, vold.size());
-    common::ErrorCode returnCode = tab->UpdateIndex(current_txn_, nkey, okey, row);
-    if (returnCode == common::ErrorCode::DUPP_KEY || returnCode == common::ErrorCode::FAILED) {
-      TIANMU_LOG(LogCtl_Level::DEBUG, "Duplicate entry: %s for primary key", vnew.data());
-      throw common::DupKeyException("Duplicate entry: " + vnew + " for primary key");
-    }
-  } else {  // common::PackType::INT
-    int64_t vnew = v.GetInt();
-    int64_t vold = GetValueInt64(row);
-    std::string_view nkey(reinterpret_cast<const char *>(&vnew), sizeof(int64_t));
-    std::string_view okey(reinterpret_cast<const char *>(&vold), sizeof(int64_t));
-    common::ErrorCode returnCode = tab->UpdateIndex(current_txn_, nkey, okey, row);
-    if (returnCode == common::ErrorCode::DUPP_KEY || returnCode == common::ErrorCode::FAILED) {
-      TIANMU_LOG(LogCtl_Level::DEBUG, "Duplicate entry :%" PRId64 " for primary key", vnew);
-      throw common::DupKeyException("Duplicate entry: " + std::to_string(vnew) + " for primary key");
-    }
-  }
-}
-
-void RCAttr::DeleteByPrimaryKey(uint64_t row, uint64_t col) {
-  auto path = m_share->owner->Path();
-  std::shared_ptr<index::RCTableIndex> tab = ha_rcengine_->GetTableIndex(path);
-  // col is not primary key
-  if (!tab)
-    return;
-  std::vector<uint> keycols = tab->KeyCols();
-  if (std::find(keycols.begin(), keycols.end(), col) == keycols.end())
-    return;
-
-  if (GetPackType() == common::PackType::STR) {
-    auto currentValue = GetValueString(row);
-    std::string_view currentRowKey(currentValue.val_, currentValue.size());
-    common::ErrorCode returnCode = tab->DeleteIndex(current_txn_, currentRowKey, row);
-    if (returnCode == common::ErrorCode::FAILED) {
-      TIANMU_LOG(LogCtl_Level::DEBUG, "Delete: %s for primary key", currentValue.GetDataBytesPointer());
-      throw common::Exception("Delete: " + currentValue.ToString() + " for primary key");
-    }
-  } else {  // common::PackType::INT
-    auto currentValue = GetValueInt64(row);
-    std::string_view currentRowKey(reinterpret_cast<const char *>(&currentValue), sizeof(int64_t));
-    common::ErrorCode returnCode = tab->DeleteIndex(current_txn_, currentRowKey, row);
-    if (returnCode == common::ErrorCode::FAILED) {
-      TIANMU_LOG(LogCtl_Level::DEBUG, "Delete: %" PRId64 " for primary key", currentValue);
-      throw common::Exception("Delete: " + std::to_string(currentValue) + " for primary key");
-    }
-  }
-}
-
-}  // namespace core
-}  // namespace Tianmu
+/* Copyright (c) 2022 StoneAtom, Inc. All rights reserved.

```

Comment:
File format changes betweent DOS and UNIX formart. Can not distingish what is the differ. Pls just commit the changed lines. You can use unix2dos or dos2unix to keep the file format consistency.

Reply:
in the future this would be take of